packages/apply/index.ts

```diff
@@ -0,0 +1,396 @@
+  // @ts-ignore TODO bumpType should be narrowed to meet ReleaseType
+  let next = semver.inc(pkg.version, bumpType);
+  if (next) {
+    pkg.version = next;
+    pkg.versionMajor = semver.major(next);
+    pkg.versionMinor = semver.minor(next);
+    pkg.versionPatch = semver.patch(next);
+  }
+  if (pkg.vfile && pkg.pkg) {
+    if (pkg.vfile.extname === ".json") {
+      // for javascript
+      // @ts-ignore TODO bumpType should be narrowed to meet ReleaseType
+      let version = semver.inc(pkg.pkg.version, bumpType);
+      if (version) pkg.pkg.version = version;
+    } else if (pkg.vfile.extname === ".toml") {
+      // for rust
+      // @ts-ignore TODO bumpType should be narrowed to meet ReleaseType
+      let version = semver.inc(pkg.pkg.package.version, bumpType);
+      // @ts-ignore TODO we need to normalize Pkg for toml? Or make some union type
+      if (version) pkg.pkg.package.version = version;
+    }
+  }
+  return pkg;
+};
+
+const bumpDeps = ({
+  packageFile,
+  dep,
+  bumpType,
+}: {
+  packageFile: PackageFile;
+  dep: string;
+  bumpType: string;
+}) => {
+  let pkg = { ...packageFile };
+
+  if (pkg.pkg && pkg.vfile && pkg.pkg.dependencies)
+    Object.keys(pkg.pkg.dependencies).forEach((existingDep) => {
+      if (
+        existingDep === dep &&
+        pkg.pkg &&
+        pkg.pkg!.dependencies &&
+        pkg.pkg!.dependencies[dep]
+      ) {
+        if (pkg.vfile!.extname === ".json") {
+          // for javascript
+          let version = semver.inc(
+            pkg.pkg.dependencies[dep],
+            // @ts-ignore TODO deal with ReleaseType
+            bumpType
+          );
+          if (version) pkg.pkg.dependencies[dep] = version;
+        } else if (pkg.vfile!.extname === ".toml") {
+          // for rust
+          if (typeof pkg.pkg.dependencies[dep] === "object") {
+            // @ts-ignore TODO deal with nest toml
+            pkg.pkg.dependencies[dep].version = incWithPartials(
+              // @ts-ignore TODO deal with nest toml
+              pkg.pkg.dependencies[dep].version,
+              bumpType
+            );
+          } else {
+            let version = incWithPartials(pkg.pkg.dependencies[dep], bumpType);
+            if (version) pkg.pkg.dependencies[dep] = version;
+          }
+        }
+      }
+    });
+
+  if (pkg.pkg && pkg.vfile && pkg.pkg.devDependencies)
+    Object.keys(pkg.pkg.devDependencies).forEach((existingDep) => {
+      if (
+        existingDep === dep &&
+        pkg.pkg &&
+        pkg.pkg!.devDependencies &&
+        pkg.pkg!.devDependencies[dep]
+      ) {
+        if (pkg.vfile!.extname === ".json") {
+          // for javascript
+          // @ts-ignore TODO deal with ReleaseType
+          let version = semver.inc(pkg.pkg.devDependencies[dep], bumpType);
+          if (version) pkg.pkg.devDependencies[dep] = version;
+        }
+      }
+    });
+
+  return pkg;
+};
+
+const incWithPartials = (version: string, bumpType: string) => {
```

Comment:
Should the `version` be its own proper type?
And the `bumpType` be an enum?

Reply:
I actually tried an enum at first, but I was having trouble getting it to work (lots of these cascade through the packages). Semver also has its own types, so that might have been part of my difficulty.

I think the answer to both of these is "yes", but for this PR, I am focusing on the same functionality and only necessary / low-effort code changes. We are looking at supporting calver as well, which means we definitely need to give this attention. I think the best path is a separate PR that focuses solely on getting the version type threaded through everything and prepping/abstracting (if needed) to support calver or other versioning schemes.

Reply:
tracking this here: https://github.com/jbolda/covector/issues/154